diff --git a/node_modules/vite/.bun-tag-6d4304d33a066bc7 b/.bun-tag-6d4304d33a066bc7
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/vite/.bun-tag-c4b4f19272a68fee b/.bun-tag-c4b4f19272a68fee
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/node/chunks/dep-DBxKXgDP.js b/dist/node/chunks/dep-DBxKXgDP.js
index 2cd82a2f105df2ac39d49cc8c8f787325d882f81..209726c2fef03e57481e1682a9f685d3e9d60872 100644
--- a/dist/node/chunks/dep-DBxKXgDP.js
+++ b/dist/node/chunks/dep-DBxKXgDP.js
@@ -49218,6 +49218,7 @@ async function loadConfigFromFile(configEnv, configFile, configRoot = process.cw
     return null;
   }
   try {
+    console.log("configLoader", configLoader);
     const resolver = configLoader === "bundle" ? bundleAndLoadConfigFile : configLoader === "runner" ? runnerImportConfigFile : nativeImportConfigFile;
     const { configExport, dependencies } = await resolver(resolvedPath);
     debug?.(`config file loaded in ${getTime()}`);
@@ -49231,6 +49232,7 @@ async function loadConfigFromFile(configEnv, configFile, configRoot = process.cw
       dependencies
     };
   } catch (e) {
+    console.log("vite error", e);
     const logger = createLogger(logLevel, { customLogger });
     checkBadCharactersInPath("The config path", resolvedPath, logger);
     logger.error(colors$1.red(`failed to load config from ${resolvedPath}`), {
@@ -49255,12 +49257,15 @@ async function runnerImportConfigFile(resolvedPath) {
 }
 async function bundleAndLoadConfigFile(resolvedPath) {
   const isESM = typeof process.versions.deno === "string" || isFilePathESM(resolvedPath);
+  console.log("isESM", isESM);
   const bundled = await bundleConfigFile(resolvedPath, isESM);
+  console.log("bundled", bundled);
   const userConfig = await loadConfigFromBundledFile(
     resolvedPath,
     bundled.code,
     isESM
   );
+  console.log("userConfig", userConfig);
   return {
     configExport: userConfig,
     dependencies: bundled.dependencies
@@ -49271,6 +49276,7 @@ async function bundleConfigFile(fileName, isESM) {
   const dirnameVarName = "__vite_injected_original_dirname";
   const filenameVarName = "__vite_injected_original_filename";
   const importMetaUrlVarName = "__vite_injected_original_import_meta_url";
+  console.log("vite before build");
   const result = await build$3({
     absWorkingDir: process.cwd(),
     entryPoints: [fileName],
@@ -49383,6 +49389,7 @@ async function bundleConfigFile(fileName, isESM) {
       }
     ]
   });
+  console.log("vite after build");
   const { text } = result.outputFiles[0];
   return {
     code: text,
@@ -49393,15 +49400,19 @@ const _require = createRequire$1(import.meta.url);
 async function loadConfigFromBundledFile(fileName, bundledCode, isESM) {
   if (isESM) {
     let nodeModulesDir = typeof process.versions.deno === "string" ? void 0 : findNearestNodeModules(path$b.dirname(fileName));
+    console.log("vite nodeModulesDir", nodeModulesDir);
     if (nodeModulesDir) {
       try {
+        console.log("vite mkdir", path$b.resolve(nodeModulesDir, ".vite-temp/"));
         await fsp.mkdir(path$b.resolve(nodeModulesDir, ".vite-temp/"), {
           recursive: true
         });
+        console.log("vite mkdir done");
       } catch (e) {
         if (e.code === "EACCES") {
           nodeModulesDir = void 0;
         } else {
+          console.error("vite mkdir error", e);
           throw e;
         }
       }
@@ -49411,12 +49422,17 @@ async function loadConfigFromBundledFile(fileName, bundledCode, isESM) {
       nodeModulesDir,
       `.vite-temp/${path$b.basename(fileName)}.${hash}.mjs`
     ) : `${fileName}.${hash}.mjs`;
+    console.log("vite tempFileName", tempFileName);
     await fsp.writeFile(tempFileName, bundledCode);
+    console.log("vite writeFile done");
     try {
+      console.log("vite import", pathToFileURL$1(tempFileName).href);
       return (await import(pathToFileURL$1(tempFileName).href)).default;
     } finally {
+      console.log("vite unlink", tempFileName);
       fs__default.unlink(tempFileName, () => {
       });
+      console.log("vite unlink done");
     }
   } else {
     const extension = path$b.extname(fileName);
